src/sql/workbench/contrib/notebook/browser/markdownToolbarActions.ts

```diff
@@ -150,6 +164,24 @@ export class MarkdownTextTransformer {
 		}
 	}
 
+	/**
+	 * Instantiate modal for use as callout when inserting Link or Image into markdown.
+	 * @param calloutStyle Style of callout passed in to determine which callout is rendered
+	 */
+	private async createCallout(calloutStyle: CalloutStyle, triggerCssSelector?: string) {
```

Comment:
Not sure I like using a selector for this - why wouldn't you pass in the HTMLElement itself?

See how the context menu service works - this is the example I think we should be following here for this : https://github.com/microsoft/azuredatastudio/blob/main/src/vs/platform/contextview/browser/contextMenuHandler.ts#L43

Reply:
I created custom button elements, added their click handlers, and added them to a custom toolbar which is only used if the user has opted in for the preview features.